### ISSUE TITLE
firstWeekdaySunday option not working properly; dates showing incorrectly.

### ISSUE THREAD

**Issue by elliotwesoff** [jkuri/ng2-datepicker#165]
This appears to be a two part bug.  

1.  The dates are not being displayed under the proper weekday.  I didn't notice this until I tried to implement the firstWeekdaySunday option. 

2.  The firstWeekdaySunday option doesn't respond to a boolean value like shown in the interface options.  

The options hash in the following block of code doesn't do anything at all.

    <ng2-datepicker
        [options]="{ firstWeekdaySunday: false }"
        [(ngModel)]="date">
    </ng2-datepicker>

However, when the firstWeekdaySunday option is provided any string, it adjusts the dates accordingly.

The calendar will display correctly (although with Sunday still being the first day) after using this block of code:

    <ng2-datepicker
        [options]="{ firstWeekdaySunday: 'false' }"
        [(ngModel)]="date">
    </ng2-datepicker>

Example of part one - displaying calendar days incorrectly:
![calendar-incorrect](https://cloud.githubusercontent.com/assets/7844060/22204839/4e929262-e174-11e6-88db-538548ccccee.png)

**Comment by jkuri** [comment]
hi @elliotwesoff I really love this styling, would you mind to make this as default and make a PR if you have time to do so please?

**Comment by elliotwesoff** [comment]
Sure thing. What about the bug though? ;)

**Comment by gaabmarquez** [comment]
Hi, i have the same bug, but i think its because the  names of days are harcoded in the HTML 
you can see this in
[ng2-datepicker.component.html](https://github.com/jkuri/ng2-datepicker/blob/master/src/ng2-datepicker/ng2-datepicker.component.html)

` <div class="datepicker-calendar-day-names">
          <span>S</span>
          <span>M</span>
          <span>T</span>
          <span>W</span>
          <span>T</span>
          <span>F</span>
          <span>S</span>
        </div>`

**Comment by elliotwesoff** [comment]
Oh yeah. Hmmmmm....

**Comment by jkuri** [comment]
in the near future I'll plan to drop `moment` and integrate `date-fns`. I will solve both issues then.

**Comment by overtureweb** [comment]
@elliotwesoff It actually is working correctly when you set the `[options]="{ firstWeekdaySunday: false }"` A value of _false_ results in the week starting on Monday. If you set it `[options]="{ firstWeekdaySunday: true }"` it will start on Sunday. You have it currently set to a 'false' string which actually evaluates as _truthy_ if you look at this property in the class constructor:
`this.firstWeekdaySunday = obj && obj.firstWeekdaySunday ? obj.firstWeekdaySunday : false;`

My recommendation to the developer would be to set the value to _true_ by default as doing so will align the calendar with the days of the week in the static html for the calendar and will presumably satisfy the vast majority of use cases.

**Comment by elliotwesoff** [comment]
@overtureweb yeah it was a bug, or perhaps just functionality that wasn't implemented yet. It looks like it was patched in commit bf8ee79d4878690178669338b4815df264754bdb. I haven't tried it since I opened this, @jkuri do you know if this has been resolved?

**Comment by elliotwesoff** [comment]
@overtureweb I know I've got a string there. I wrote that to illustrate the strange behavior.  Of course it shouldn't work when you use a string, but it does... sort of. I'll update my datepicker and try it again.  If it works I'll close the issue.

**Comment by kurotsukikaitou** [comment]
@jkuri The default theme is nice. But the day title header's (S M T W....) color are too dark rendering them nearly unreadable which can be seen here:
<img width="251" alt="screen shot 2017-08-23 at 9 39 32 pm" src="https://user-images.githubusercontent.com/5329246/29621690-ad73b53a-884b-11e7-97f8-a6f45b263dd3.png">
